packages/cli-kit/src/public/node/terminal-layout.ts

```diff
@@ -0,0 +1,68 @@
+const MIN_TERMINAL_WIDTH = 80
+const TERMINAL_WIDTH_RATIO = 2 / 3
+const TRUNCATION_SUFFIX_LENGTH = 3
+const CURSOR_UP_AND_CLEAR = '\x1b[1A\x1b[2K\r'
+
+/**
+ * Calculates the length of a string, ignoring ANSI escape codes.
+ * (That lets use calculate how much space the string will actually take up when displayed.).
+ *
+ * @param text - The text to measure.
+ * @returns The length of the text without ANSI escape codes.
+ */
+function getVisualLength(text: string): number {
+  // eslint-disable-next-line no-control-regex
+  return text.replace(/\x1b\[[0-9;]*m/g, '').length
+}
+
+/**
+ * Creates right-aligned text by padding between left and right text.
+ *
+ * @param leftText - The text to display on the left.
```

Comment:
is this the `Importing...` text from your image in the pr description?

Reply:
Yeah!